docs: fix missing information in createComponent guide

hiepxanh opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #51492


## What is the new behavior?

just update docs

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

JeanMeche committed:
refactor(core): Use intersections on branded types. (#49702)

An Intersection on a branded type allows us to remove some unecessary type assertions.

PR Close #49702

josephperrott committed:
ci: migrate monitoring jobs to GHA (#51350)

Migrate monitoring jobs to Github Actions from CircleCI

PR Close #51350

josephperrott committed:
docs: remove unreachable/incorrect contributor links (#51350)

PR Close #51350

JeanMeche committed:
refactor(animations): cleanup ahead of animation lazy loading (#51249)

There are no functional changes.

PR Close #51249

angular-robot committed:
build: update dependency google-closure-compiler to v20230802 (#51265)

See associated pull request for more information.

PR Close #51265

josephperrott committed:
ci: migrate zonejs tests to gha (#51352)

Migreate zonejs tests from CircleCI to Github Actions

PR Close #51352

luishcastroc committed:
docs(core): clarify InjectionToken usage (#51386)

- Emphasized the importance of using the same InjectionToken instance for both provider and injection call.
- Added examples to illustrate correct usages to prevent NullInjectorError.

PR Close #51386

bencodezen committed:
docs: migrate security guide and example to standalone (#51446)

PR Close #51446

MeAkib committed:
docs(docs-infra): for accessibility and responsive,  iframe should be in video container, contain some space horizontally, and title (#51475)

PR Close #51475

JeanMeche reviewed (approved):
You opened the PR on your repo, not the angular repo ;)

wardbell committed:
docs: Migrate Observables guides & code examples to standalone (#51516)

None of the guide pages mentions ngModules. Only `observables-in-angular` needed conversion to Standalone.

However, some of the guide pages reflect old versions of RxJS, including signatures that are no longer valid. These have been corrected.

More significantly, *the existing guide is pretty bad at explaining RxJS and its usage*. It was written (by me I think) in the very early days of Angular and Angular RxJS instruction. I've taught numerous "RxJS in Angular" classes since and learned from that experience what does and does not work with students.

There was neither the time nor the charter to completely overhaul this guide. But this commit attempts to remove what flops with students and to bring the teaching closer to what seems more effectively. I hope reviewers agree that my revisions are an improvement.

**Revised Overview**

The overview doc, `observables.md`, had a few errors (ex: `next` is NOT REQUIRED) and deprecated patterns (you now must pass the Observer object to `subscribe`).

More importantly, it was wildly overcomplicated and scary, especially when it got into multi-casting.

Moved the multi-casting section to  "RxJS Library" and rewrote it (with working example) for simplicity and context.

I made other changes in an effort to make this an overview that is  more comprehensive and more clear. I paid particular attention to the "Basic usage and terms" section.

Finally, I relocated the "Naming conventions for observables" section here from `rx-library`. This is the section that describes the dollar-sign convention. It made more sense for it to be here.

**Revised "RxJS Library" page and code**

*RxJS no longer supports chaining* and hasn't for a very long time. Removed note in `rx-library.md` that suggested you could use operator chaining.

The RxJS `pipe` discussion in the "Operators" section was just weird. Almost no one calls the `pipe` function. We certainly should *start* there. We should start with how people actually use operators - by adding them to the argument list of the `Observable.pipe()` method.

I kept the original `pipe` function example but subordinated it in a "callout". Most readers will (and should) ignore it.

`Subject` is a *critically important RxJS mechanism for creating custom observables*. It was completely missing from the list of observable creators on this guide page. So I added it to the "Observable creation functions" section of the guide and wrote an accompanying `MessageService` code sample (see the new `rx-library/app/` folder).

The `MessageService` is a pretty common pattern in Angular apps - far more common than creating an observable from a counter or an event, two of the creation patterns currently on this page.

This new section also afforded an opportunity to show how RxJS helps with building loosely coupled applications. We will soon be talking about Signals. Many will wonder whether and when they should still use RxJS.

At least a partial answer is that RxJS is really good at progressively combining and enhancing streams of data as they cross component boundaries. Of course you can pass signals around; but they are not as rich in transformers as RxJS. This is where RxJS shines.

**Revised "Comparing observables"**

The Promises section in `comparing-observables.md` had many errors and misleading remarks.

The comparison of error handling was especially egregious; the code example for that was nonsense.

The "Chain" sub-section was really about transforming values. It also failed to demonstrate chaining promise `.then`s.

Reworked these sub-sections and improved the code samples to match.

PR Close #51516

devversion committed:
build: increase idle timeout for saucelabs daemon (#51533)

Saucelabs seemingly has increased the maximum idle timeout. So we
leverage that to improve stability of our tests. Useful when e.g.
the heartbeat webdriver commands are for some reasons delayed.

PR Close #51533

devversion committed:
build: improve debug logging of saucelabs daemon (#51533)

We should print more specific details about a browser being used
for tests. This will help us debug timeout/flakiness easier.

PR Close #51533

devversion committed:
build: enable video recording for saucelabs bazel tests (#51533)

We temporarily enable video recording for Saucelabs bazel tests. We are
seeing some flakiness here, and before we can replace the legacy job,
we should understand why the browsers sometimes disconnect.

PR Close #51533

bencodezen committed:
docs: migrate user-input code example to standalone (#51466)

PR Close #51466

devknoll committed:
refactor(core): make AfterRenderEventManager tree-shakable (#51541)

In preparation for adding support for phases to after*Render, which will increase the implementation size, this commit splits out the optional logic so that it can be tree-shaken and dynamically loaded.

PR Close #51541

atscott committed:
docs: update router reference to use standalone (#51560)

this commit updates the router reference to use standalone features.

PR Close #51560

atscott committed:
fix(router): Ensure title observable gets latest values (#51561)

The data `Observable` is not updated unless there have been changes to
the object. The current diffing does not look at `symbol` keys of the
object but the `title` property is stored as a private `symbol`. This
commit updates the object diffing to include symbols.

fixes #51401

PR Close #51561

alan-agius4 committed:
feat(platform-browser): enable removal of styles on component destroy by default (#51571)

This change aligns the settings between G3 and P3 as `REMOVE_STYLES_ON_COMPONENT_DESTROY_DEFAULT` is already set to `true` internally.

BREAKING CHANGE: `REMOVE_STYLES_ON_COMPONENT_DESTROY` default value is now `true`. This causes CSS of components to be removed from the DOM when destroyed. You retain the previous behaviour by providing the `REMOVE_STYLES_ON_COMPONENT_DESTROY` injection token.

```ts
import {REMOVE_STYLES_ON_COMPONENT_DESTROY} from '@angular/platform-browser';
...
providers: [{
  provide: REMOVE_STYLES_ON_COMPONENT_DESTROY,
  useValue: false,
}]
```

PR Close #51571

bencodezen committed:
docs: migrate two-way-binding to standalone (#51574)

PR Close #51574

angular-robot committed:
build: update actions/checkout digest to f43a0e5 (#51576)

See associated pull request for more information.

PR Close #51576

bencodezen committed:
docs: migrate toh-pt0 to standalone (#51575)

PR Close #51575

thePunderWoman committed:
docs: release notes for the v16.2.3 release

thePunderWoman committed:
release: cut the v17.0.0-next.2 release

josephperrott committed:
refactor(docs-infra): correct incomplete string replacement (#51585)

Find all usages of `\n` and replace them instead of just the first.

PR Close #51585

JeanMeche committed:
docs: remove NG6999 error page. (#51588)

`NGMODULE_VE_DEPENDENCY_ON_IVY_LIB` was a ViewEngine related error. This commit removes the doc page but keeps a redirection for older versions still throwing this error.

PR Close #51588

AleksanderBodurri committed:
refactor(devtools): migrate material legacy components to mdc (#51590)

Removes all legacy angular material imports and moves devtools over to mdc.

PR Close #51590

josephperrott committed:
ci: fix permissions issue in zonejs ci test (#51600)

Correct permissions issue in zonejs folders

PR Close #51600

josephperrott committed:
fix(docs-infra): fix incomplete escaping (#51604)

Correct incomplete escaping

PR Close #51604

angular-robot committed:
build: update dependency archiver to v6 (#51605)

See associated pull request for more information.

PR Close #51605

hiepxanh committed:
docs: fix missing information in createComponent
